build.proj

```diff
@@ -62,7 +63,20 @@
     <MSBuild Projects="@(PkgProject)"
              Targets="Pack" />
   </Target>
-
+  
+  <ItemGroup>
+    <TestFile Include="$(MSBuildThisFileDirectory)/test/data/external/winequality-white.csv" Url="https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-white.csv">
```

Comment:
Nit, I believe the attribute and subelement formats are equivalent, you might want to pick one or the other.

Reply:
I can't move Include to subelement, but I can move Url to it. having everything as attribute is looks too long

---
In reply to: [189065564](https://github.com/dotnet/machinelearning/pull/170#discussion_r189065564) [](ancestors = 189065564)

Reply:
I'm not a requesting change here, just giving some information for future reference.

The `Include` attribute on an MSBuild Item is "special".  It has to be an XML attribute.  Before MSBuild v15, all "custom" metadata (like your Url attribute) could NOT be put as an XML attribute - instead it had to be a sub-element.  In v15, they changed it so custom metadata now can be in either an attribute or sub-element.

A format I like and use regularly is the following:

```XML
    <TestFile Include="$(MSBuildThisFileDirectory)/test/data/external/winequality-white.csv"
              Url="https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-white.csv"
              DestinationFile="$(MSBuildThisFileDirectory)test/data/external/winequality-white.csv" />
```

I think it is compact, reads easy, and is easy to maintain/update.